### PR TITLE
some more document on dependency order

### PR DESCRIPTION
**Pull request by scopatz**
from https://github.com/conda-forge/glew-feedstock/issues/9

### CODE REVIEW

File: src/recipe.rst [hunk 1]
```diff
@@ -86,15 +86,23 @@ Avoid Dependencies Outside of Conda-Forge
...
-If your dependencies do not change with the python version, or with the
-platform, consider making your build :ref:`noarch <noarch>`, this will allow the recipe to build faster, and free some CI resources for other projects.
-
+of ABI compatibility for the conda-forge packages.
+**Only in extreme cases should you rely on a dependency outside of conda-forge.**
+In these cases, the basic heirarchy for dependencies, in order of preference, is,
+
+1. Make or use a conda-forge package for the dependency
+2. Use a defaults package for a dependency
+3. Use a CDT for a dependency
```
Comment (sodre) [inline src/recipe.rst]: @scopatz could you add a link and/or expand the CDT acronym?

**Comment by jakirkham** [comment]
As often this comes up in the context of X11 packages, I wonder if we should say something specifically about conda-forge X11 packages as well. Can completely understand if we don't want to handle this point in this PR. Just wanted to get a sense of people's thoughts on these changes.

**Comment by scopatz** [comment]
I think if we wanted to address X11 that could be a different PR.  Not sure I really understand all of the subtlety personally of the X11 space.

File: src/recipe.rst [hunk 2]
```diff
@@ -86,15 +86,23 @@ Avoid Dependencies Outside of Conda-Forge
...
-
+of ABI compatibility for the conda-forge packages.
+**Only in extreme cases should you rely on a dependency outside of conda-forge.**
+In these cases, the basic heirarchy for dependencies, in order of preference, is,
+
+1. Make or use a conda-forge package for the dependency
+2. Use a defaults package for a dependency
+3. Use a `CDT <https://conda.io/docs/user-guide/tasks/build-packages/compiler-tools.html#cdt-packages>`_ for a dependency
+4. Use a ``yum_requirements.txt`` for a dependency (deprecated)
+5. Silently require the package to exist on the user's system
```
[the remark this answers is not in the document]
Comment (scopatz) [reply src/recipe.rst]: Updated!

**Comment by scopatz** [comment]
This is ready for review again!

**Comment by scopatz** [comment]
Is there anything else that needs to go in here?

**Pull request merged**